include/multipass/cli/prompters.h

```diff
@@ -55,6 +55,23 @@ class PlainPrompter : public BasePrompter
     std::string prompt(const std::string&) const override;
 };
 
+class PassphrasePrompter : public BasePrompter
+{
+public:
+    explicit PassphrasePrompter(Terminal* term);
+    virtual ~PassphrasePrompter();
+
+    std::string prompt(const std::string&) const override;
+};
+
+class NewPassphrasePrompter : public PassphrasePrompter
+{
+public:
+    using PassphrasePrompter::PassphrasePrompter;
+    using PassphrasePrompter::prompt;
+
+    std::string prompt(const std::string&, const std::string&) const;
```

Comment:
Not sold on having a special interface here :/

Maybe we need to repurpose the argument to `prompt()` to be a piece of the actual prompt printed, and then the prompter has a standard format for it?

Reply:
Ok, I refactored this quite a bit.  Let me know what you think.